worlds/pokemon_crystal/options.py

```diff
@@ -0,0 +1,374 @@
+        Forces Goal to Elite Four unless Silver Cave is included.
+        Goal badges will be limited to 8 if badges are shuffled or vanilla"""
+    display_name = "Johto Only"
+    default = 0
+    option_off = 0
+    option_on = 1
+    option_include_silver_cave = 2
+
+
+class EliteFourBadges(Range):
+    """Number of badges required to enter Victory Road"""
+    display_name = "Elite Four Badges"
+    default = 8
+    range_start = 1
+    range_end = 16
+
+
+class RedBadges(Range):
+    """Number of badges required to open Silver Cave"""
+    display_name = "Red Badges"
+    default = 16
+    range_start = 1
+    range_end = 16
+
+
+class RandomizeBadges(Choice):
+    """Shuffles badges into the pool"""
+    display_name = "Randomize Badges"
+    default = 2
+    option_vanilla = 0
+    option_shuffle = 1
+    option_completely_random = 2
+
+
+class RandomizeHiddenItems(Toggle):
+    """Shuffles hidden items into the pool"""
+    display_name = "Randomize Hidden Items"
+    default = 0
+
+
+class RequireItemfinder(DefaultOnToggle):
+    """Hidden items require Itemfinder in logic"""
+    display_name = "Require Itemfinder"
+
+
+class Trainersanity(Toggle):
+    """Adds checks for defeating trainers"""
+    display_name = "Trainersanity"
+    default = 0
+
+
+class TrainersanityAlerts(Choice):
+    """Shows a message box or plays a sound for trainersanity checks"""
+    display_name = "Trainersanity"
+    default = 1
+    option_no_alerts = 0
+    option_message_box = 1
+    option_sound_only = 2
+
+
+class RandomizePokegear(Toggle):
+    """Shuffles the Pokegear and cards into the pool"""
+    display_name = "Randomize Pokegear"
+    default = 0
+
+
+class RandomizeBerryTrees(Toggle):
+    """Shuffles berry tree items into the pool"""
+    display_name = "Randomize Berry Trees"
+    default = 0
+
+
+class RandomizeStarters(Choice):
+    """Randomizes species of starter Pokemon"""
+    display_name = "Randomize Starters"
+    default = 0
+    option_vanilla = 0
+    option_unevolved_only = 1
+    option_completely_random = 2
+
+
+class RandomizeWilds(Toggle):
+    """Randomizes species of wild Pokemon"""
+    display_name = "Randomize Wilds"
+    default = 0
+
+
+class NormalizeEncounterRates(Toggle):
+    """Normalizes chance of encountering each wild Pokemon slot"""
+    display_name = "Normalize Encounter Rates"
+    default = 0
+
+
+class RandomizeStaticPokemon(Toggle):
+    """Randomizes species of static Pokemon encounters"""
+    display_name = "Randomize Static Pokemon"
+
+
+class RandomizeTrainerParties(Choice):
+    """Randomizes Pokemon in enemy trainer parties"""
+    display_name = "Randomize Trainer Parties"
+    default = 0
+    option_vanilla = 0
+    option_match_types = 1
+    option_completely_random = 2
+
+
+class RandomizeLearnsets(Choice):
+    """vanilla: Vanilla movesets
+    randomize: Random movesets
+    start_with_four_moves: Random movesets with 4 starting moves"""
```

Comment:
Most games use the WebHost/website displays here instead of the yaml name. Could make the same change a few other places, including "randomize" in this option
```suggestion
    "Start With Four Moves": Random movesets with 4 starting moves"""
```